storage2/src/state.rs

```diff
@@ -15,7 +19,8 @@ use crate::snapshot::Snapshot;
 /// implemented as a RYW cache over a pinned JMT version.
 pub struct State {
     snapshot: Arc<Snapshot>,
-    unwritten_changes: HashMap<String, Vec<u8>>,
+    // A `None` value represents deletion.
+    pub(crate) unwritten_changes: HashMap<String, Option<Vec<u8>>>,
```

Comment:
Rather than a `HashMap`, I think we should use a `BTreeMap` here, so that in the future, we'll be able to iterate over keys.